src/onnx/parse_softmaxcrossentropyloss.cpp

```diff
@@ -0,0 +1,420 @@
+
+        // Meta parameters based on input scores shape
+        size_t ndims = scores_shape.ndim();
+        size_t class_size = scores_shape.lens().at(1);
+        bool is_k_dim  = (ndims > 3);
+
+        // Ignore_index is optional attribute, assign this as a scalar literal input to the op
+        instruction_ref ignore_index;
+        auto has_ignore_index =
+            normalize_input_index(parser, info, static_cast<int64_t>(class_size), ignore_index);
+
+        bool has_weights = (args.size() > 2);
+        instruction_ref weights = get_weights(info, args, scores_shape, class_size);
+
+        // Adjust weights based on ignore index if that's set to reduce output after mul to zero
+        // Saves us from doing a where() here and just scale at the end
+        if(has_ignore_index)
+        {
+            auto weights_shape = weights->get_shape();
+            std::vector<float> zero_val_vect(weights_shape.elements(), 0);
+            auto zero_val = info.add_literal(migraphx::literal(weights_shape, zero_val_vect));
+            weights       = info.add_instruction(
+                migraphx::make_op("scatter_none", {{"axis", 0}}), weights, ignore_index, zero_val);
+        }
+
+        // Need to perform softmax on all the data before we select final axes
+        scores =
+            info.add_instruction(migraphx::make_op("softmax", {{"axis", 1}}), scores);
+
+        // Index selection before loss calculation completed
+        auto gathernd_indicies = handle_index_selection(info, labels);
+
+        std::vector<int64_t> perm(class_size, 0);
+        if (is_k_dim)
+        {
+            std::iota(++perm.begin(), perm.end(), 2);
+            perm.at(class_size - 1) = 1;
+            scores = info.add_instruction(migraphx::make_op("transpose", {{"permutation", perm}}), scores);
+        }
+
+        scores = info.add_instruction(migraphx::make_op("gathernd"), scores, gathernd_indicies);
+
+        std::vector<int64_t> axis_list(ndims-1, 0);
+        std::iota(++(axis_list.begin()), axis_list.end(), 2);
+        weights = info.add_instruction(migraphx::make_op("unsqueeze", {{"axes", axis_list}}), weights);
+        weights = info.add_instruction(migraphx::make_op("multibroadcast", {{"out_lens", scores_shape.lens()}}), weights);
+        if (is_k_dim)
+            weights = info.add_instruction(migraphx::make_op("transpose", {{"permutation", perm}}), weights);
+        weights = info.add_instruction(migraphx::make_op("gathernd"), weights, gathernd_indicies);
+
+        // Do pointwise operators on the final set of indicies and scores we care about rather than
+        // before so that we're not doing a bunch of pointwise on items that aren't part of the loss calulation.
+        // Helps more in the mismatched Batch/Class and multi batch size cases. 
+        auto log_sm_scores  = info.add_instruction(migraphx::make_op("log"), scores);
+        auto neg_lsm_scores = info.add_instruction(migraphx::make_op("neg"), log_sm_scores);
```

Comment:
Please document the formulae that is being used here in this operator, to calculate the cross entropy. Thanks.

Reply:
Sure. I'll add as comment